operator/pkg/controllers/master/kubeconfigs.go

```diff
@@ -25,18 +25,34 @@ import (
 	"github.com/awslabs/kit/operator/pkg/utils/secrets"
 	"go.uber.org/zap"
 
-	"gopkg.in/yaml.v2"
 	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	clientcmdlatest "k8s.io/client-go/tools/clientcmd/api/latest"
 	certutil "k8s.io/client-go/util/cert"
 )
 
+const (
+	localhostEndpoint = "127.0.0.1"
+)
+
 // reconcileKubeConfigs creates required kube configs and stores them as secret in api server
 func (c *Controller) reconcileKubeConfigs(ctx context.Context, controlPlane *v1alpha1.ControlPlane) error {
-	for _, request := range []*secrets.Request{
-		kubeAdminCertConfig(controlPlane.ClusterName()),
-		kubeSchedulerCertConfig(controlPlane.ClusterName()),
-		kubeControllerManagerCertConfig(controlPlane.ClusterName()),
+	// kubeconfig certs are signed by master root CA certs, if the root CA is not found return
+	caSecret, err := c.keypairs.GetSecretFromServer(ctx,
+		object.NamespacedName(rootCASecretNameFor(controlPlane.ClusterName()), controlPlane.Namespace))
+	if err != nil {
+		return err
+	}
+	endpoint, err := c.getClusterEndpoint(ctx, object.NamespacedName(
+		controlPlane.ClusterName(), controlPlane.Namespace))
+	if err != nil {
+		return err
```

Comment:
Can you cut an issue to deep dive error bubbling to make sure that error messages have enough context?

Reply:
https://github.com/awslabs/kubernetes-iteration-toolkit/issues/26